hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetFilterQueryManagement.java

```diff
@@ -279,10 +280,16 @@ public TargetFilterQuery updateAutoAssignDS(final AutoAssignDistributionSetUpdat
             targetFilterQuery.setAutoAssignInitiatedBy(tenantAware.getCurrentUsername());
             targetFilterQuery.setAutoAssignActionType(sanitizeAutoAssignActionType(update.getActionType()));
             targetFilterQuery.setAutoAssignWeight(update.getWeight());
-            targetFilterQuery.setConfirmationRequired(update.getConfirmationRequired());
+            final boolean confirmationRequired = update.isConfirmationRequired() == null ? isUserConsentEnabled()
+                    : update.isConfirmationRequired();
+            targetFilterQuery.setConfirmationRequired(confirmationRequired);
         }
         return targetFilterQueryRepository.save(targetFilterQuery);
     }
+    
+    private boolean isUserConsentEnabled(){
```

Comment:
Wondering about the name (which reflects one use case). would have expected sth like "isConfirmationSupportEnabled" or "isAssignmentConfirmationsEnabled"...

Reply:
Will be discussed after implementing all features of the user consent story